.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,12 @@
+<!--
+
+Note
+
+* Mark the PR as draft until it's ready to be reviewed.
+* Please update the documentation to reflect the changes made in the PR.
+* If the command is covered by tests under ./tests, please add/update tests for any changes unless you have a good reason. 
+* Make a new commit to resolve conversations instead of `push -f`.
+* To resolve merge conflicts, merge master instead of rebasing.
```

Comment:
```suggestion
* To resolve merge conflicts, merge from the `main` branch instead of rebasing over `main`.
```

Reply:
Once the merged from `main`, the commit tree will becomes messing. Is there a good practice here?

Reply:
@vanpipy There is something to be said about just merging instead of rebasing. But, looking at the [commit history](https://github.com/tj/git-extras/commits/main/), of this project, the Change Requests are _usually_ "Squashed and Merged" into `main` by the maintainers (unless they are not). So, in the common case, the commit tree will not become messed up

Reply:
I got the idea of that the "Squashed and Merged" is the fallback to squashed anyways to take us away from something disaster when using GitHub. Great 😆